ably-ios/ARTAuth.m

```diff
@@ -38,12 +40,17 @@ - (instancetype)init:(ARTRest *)rest withOptions:(ARTClientOptions *)options {
         } else if (options.tokenDetails) {
             [self.logger debug:@"ARTAuth: setting up auth method Token with supplied token only"];
             _method = ARTAuthMethodToken;
+        } else if (options.authUrl && options.authCallback) {
+            [NSException raise:@"ARTAuthException" format:@"Could not setup authentication method with given options."];
```

Comment:
Can this rather say "Incompatible authentication configuration: please specify either authCallback and authUrl"